src/libs/actions/TaxRate.ts

```diff
@@ -119,10 +182,299 @@ function clearTaxRateError(policyID: string, taxID: string, pendingAction?: Pend
+                            acc[taxID] = {
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+                                errors: ErrorUtils.getMicroSecondOnyxError('workspace.taxes.errors.deleteFailureMessage'),
+                            };
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters = {
+        policyID,
+        taxNames: JSON.stringify(taxesToDelete.map((taxID) => policyTaxRates[taxID].name)),
+    } satisfies DeletePolicyTaxesParams;
+
+    API.write(WRITE_COMMANDS.DELETE_POLICY_TAXES, parameters, onyxData);
+}
+
+/**
+ * Rename policy tax
+ */
+function updatePolicyTaxValue(policyID: string, taxID: string, taxValue: number) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+    const originalTaxRate = {...policy?.taxRates?.taxes[taxID]};
+    const stringTaxValue = `${taxValue}%`;
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxID]: {
+                                value: stringTaxValue,
+                                pendingFields: {value: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
+                                errorFields: {value: null},
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxID]: {value: stringTaxValue, pendingFields: {value: null}, errorFields: {value: null}},
+                        },
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: {
+                            [taxID]: {
+                                value: originalTaxRate.value,
+                                pendingFields: {value: null},
+                                errorFields: {value: ErrorUtils.getMicroSecondOnyxError('workspace.taxes.errors.updateFailureMessage')},
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+    };
+
+    if (!originalTaxRate.name) {
+        throw new Error('Tax rate name not found');
+    }
```

Comment:
Why do we perform this check here?

Reply:
Before we called the API with the name, not the ID. I forgot to remove it.